service/src/scripts/approve_utils.ts

```diff
@@ -0,0 +1,69 @@
+import 'dotenv/config'
+import { Web3Interface } from './web3-interface'
+import { BigNumber, ethers } from 'ethers'
+import { exit } from 'process'
+import { RPC_EXCEPTION } from '../utils/exit-codes'
+import { archLogger } from '../utils/chalk-theme'
+import createPrompt from 'prompt-sync';
+
+const prompt = createPrompt({ sigint: true });
+
+const _hasAllowance = async (web3Interface: Web3Interface, amt: BigNumber) => {
+  const allowance = await web3Interface.sarcoToken.allowance(web3Interface.ethWallet.address, process.env.SARCO_DIAMOND_ADDRESS!);
+
+  if (allowance.gte(amt)) {
+    archLogger.notice("Already approved");
+    return true;
+  }
+
+  return false;
+}
+
+/**
+ * Approves Sarcophagus contracts' spending SARCO on the connected account's behalf up to `MaxUint256` tokens.
+ * 
+ * `runApprove` is really only used by the approve script - which IS attempting to approve `MaxUint256` tokens -
+ * or by `requestApproval` when there's either not enough allowance or the previous `MaxUint256` allowance has
+ * somehow dropped so low that the pending transfer exceeds that allowance. So checking for allowance on `MaxUint256`
+ * is exactly what is wanted.
+ */
+export const runApprove = async (web3Interface: Web3Interface) => {
+  archLogger.notice("Approving Sarcophagus contracts to spend SARCO on your behalf...");
+
+  if (await _hasAllowance(web3Interface, ethers.constants.MaxUint256)) {
+    return;
+  }
+
+  setInterval(() => process.stdout.write("."), 1000);
+
+  const handleException = (e) => {
+    archLogger.error(e);
+    exit(RPC_EXCEPTION);
+  }
+
+  const tx = await web3Interface.sarcoToken.approve(process.env.SARCO_DIAMOND_ADDRESS!, ethers.constants.MaxUint256).catch(handleException);
+
+  archLogger.info("Waiting for transaction")
+  tx.wait().then(() => {
+    archLogger.notice("SUCCESS!");
+  }).catch(handleException);
+
+  return tx;
+}
```

Comment:
Ah, I'm on the fence on this idea, although that would be helpful

Reply:
we wouldn't want to do this for the unwrapping TX, since that  will be automatic (and not run via a command), we wouldn't want to stop that TX from getting executed (even though there may be a high chance it would fail).  But for any TX they manually run from the command line, it might be a good way to save the archaeologist money by giving them a warning before executing.